src/rdkafka.c

```diff
@@ -4078,6 +4078,10 @@ rd_kafka_op_res_t rd_kafka_poll_cb(rd_kafka_t *rk,
                 rd_kafka_purge(rk, rko->rko_u.purge.flags);
                 break;
 
+        case RD_KAFKA_OP_METADATA_UPDATE:
+                rd_kafka_metadata_update_op(rk, rko->rko_u.metadata.mdi);
```

Comment:
RD_KAFKA_OP_RES_HANDLED must be returned otherwise it's not destroyed. The decision on the fields to use should be done by the handling code
```suggestion
                res = rd_kafka_metadata_update_op(rk, rko);
```

Reply:
Updated

Reply:
I'm returning `RD_KAFKA_OP_RES_HANDLED` from `rd_kafka_metadata_update_op` not sure if that's correct